chassembly/src/main/java/org/quiltmc/chasm/lang/api/ast/BinaryNode.java

```diff
@@ -287,6 +287,10 @@ public Node evaluate(Evaluator evaluator) {
             }
             break;
             case EQUAL: {
+                if (left instanceof NullNode ^ right instanceof NullNode) {
```

Comment:
These Xor operations don't feel good to me.
While they are correct, I find them hard to understand when reading it.
I'd suggest using 2 if statements instead of the (general) form `if (left instanceof NullNode) return right instanceof NullNode;` or something similar.